packages/twenty-server/src/workspace/messaging/services/gmail-partial-sync.service.ts

```diff
@@ -207,12 +244,33 @@ export class GmailPartialSyncService {
         maxResults,
       });
 
-      return { history: history.data };
+      let nextPageToken = history.data.nextPageToken;
```

Comment:
Be careful, history can be empty, I've merged a PR today that should handle that you might just need to rebase